eZ/Publish/API/Repository/FieldType.php

```diff
@@ -228,8 +228,9 @@ public function validateFieldSettings($fieldSettings);
      *
      * @param \eZ\Publish\API\Repository\Values\ContentType\FieldDefinition $fieldDef The field definition of the field
      * @param \eZ\Publish\SPI\FieldType\Value $value The field value for which an action is performed
+     * @param array|null $fieldSettings Actual settings of the field
      *
      * @return \eZ\Publish\SPI\FieldType\ValidationError[]
      */
-    public function validateValue(FieldDefinition $fieldDef, Value $value);
+    public function validateValue(FieldDefinition $fieldDef, Value $value, array $fieldSettings = null);
```

Comment:
Maybe I am missing the evident here, but why not use the settings from within the FieldDefinition  ?

Reply:
Setting gathered from $fieldDef contain old setting and no the new one

Reply:
Sorry I am at a loss here.

The problem mentioned in the original ticket happens when a new user is being created, not when the User ContentType is being edited.
So I see no 'old and new field settings' at play here, but only the 'currrent settings for the current field', that are stored in the FieldDefinition...

Reply:
The point is when you create new Field Type, and add new field to it (for example Country field) you have default setting 'Multiple choice' set to false. When you want to change this setting and also set two or more default countries, you can not because `$fieldDefinition->getFieldSettings()` returns `array:1 [▼
  "isMultiple" => false
]`. Not just set setting. You can see this behavior in admin v2

Reply:
@gggeek The problem here was wrong issue number :)

Side: I'm +0.5 with the change, I just need to think carefully about BC here (which includes looking over codebase). I'll try to do it shortly.

Reply:
Aha now I understand it better.
So the problem seems to come from the fact that setting a default value for a content type goes through the same validation as setting a value for a content would.

I am still -1 with the proposed fix though, as it really makes the API look bad.

Why not try instead to revert the order of application of the changes?
Whenever an operation tries to update at the same time both field settings and field default value, first validate the settings, then with those generate a new fieldDefinition, and with the new FieldDefinition validate the new default value.

On a slightly more philosophical level: I am always way of designing APIs that follow the needs of any single interface, be it rest-api, admin v1 or admin v5. By experience, whenever you do that without extremely careful thinking, you end up with an API that is not general enough and later impedes evolution.
Even though ez5 (and up) are much better than ez4 in this regard, I have found more than one case were the API was badly designed because the developers thought mostly about the needs of the rest-api - you can find those cases as bugs that I opened against the ez kernel and were discovered while trying to implement a full interface on top of the kernel api in the kaliop migration bundle.